examples/calendar/backend/services/users/controller.ts

```diff
@@ -0,0 +1,58 @@
+import { NextFunction, Request, Response } from "express";
+
+import { User } from "../../models/user.model";
+
+export const find = (req: Request, res: Response, next: NextFunction) => {
+  // If a query string ?publicAddress=... is given, then filter results
+  const whereClause =
+    req.query && req.query.publicAddress
+      ? {
+          where: { publicAddress: req.query.publicAddress },
+        }
+      : undefined;
+
+  return User.findAll(whereClause)
+    .then((users: User[]) => res.json(users))
+    .catch(next);
+};
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+  // AccessToken payload is in req.user.payload, especially its `id` field
+  // UserId is the param in /users/:userId
+  // We only allow user accessing herself, i.e. require payload.id==userId
+  if ((req as any).user.payload.id !== +req.params.userId) {
+    return res.status(401).send({ error: "You can can only access yourself" });
+  }
+  return User.findByPk(req.params.userId)
+    .then((user: User | null) => res.json(user))
+    .catch(next);
+};
+
+export const create = (req: Request, res: Response, next: NextFunction) =>
+  User.create(req.body)
+    .then((user: User) => res.json(user))
+    .catch(next);
+
+export const patch = (req: Request, res: Response, next: NextFunction) => {
+  // Only allow to fetch current user
+  if ((req as any).user.payload.id !== +req.params.userId) {
+    return res.status(401).send({ error: "You can can only access yourself" });
+  }
+  return User.findByPk(req.params.userId)
+    .then((user: User | null) => {
+      if (!user) {
+        return user;
+      }
+
+      Object.assign(user, req.body);
```

Comment:
i didn't really understand why are we doing this?

Reply:
I have removed this code. It was just a patch from other work.